### PR TITLE
[CI] Fix failing arm build

### PR DESCRIPTION
**Pull request by hcho3**

### CODE REVIEW

File: tests/ci_build/build_via_cmake.sh [hunk 1]
```diff
@@ -7,6 +7,13 @@ then
   echo "Activating Conda environment ${conda_env}"
   shift 1
   cmake_args="$@"
+
+  # Workaround for file permission error
+  if [[ -n $CI_BUILD_UID ]]
+  then
+    gosu root chown -R "${CI_BUILD_UID}:${CI_BUILD_GID}" /opt/python/envs
```
Comment (hcho3) [inline tests/ci_build/build_via_cmake.sh:14]: This may increase our test time.

**Comment by trivialfis** [comment]
For some reason, the tests on PR seem to be passing now? https://github.com/dmlc/xgboost/pull/8754

**Comment by hcho3** [comment]
Permission issue may come up again, since some code in the wild assumes that the Conda env is writable by the user. Please review this PR @trivialfis

**Review by trivialfis — APPROVED** [review]
Thank you for the fix!

**Comment by trivialfis** [comment]
> Permission issue may come up again, since some code in the wild assumes that the Conda env is writable by the user

Did you happen to figure out who was doing this? I find it uncomfortable that someone's writing to my software installation directory.

**Comment by hcho3** [comment]
@trivialfis 

> Did you happen to find out who was doing this?

No idea. I could not reproduce the error on my end.

> I would find it uncomfortable that someone's writing to my software installation directory.

Beats me. For now, `chown` is fast enough. I am merging this PR to unblock the CI.

**Pull request merged**